extension/observer/k8sobserver/extension.go

```diff
@@ -30,23 +33,42 @@ type k8sObserver struct {
 	informer cache.SharedInformer
 	stop     chan struct{}
 	config   *Config
+	server   *http.Server
 }
 
 func (k *k8sObserver) Start(ctx context.Context, host component.Host) error {
 	go k.informer.Run(k.stop)
+
+	// Use custom status port until there's a way to export status via zpages or something else.
+	if k.config.StatusAddr != "" {
```

Comment:
This appears to be unrelated to the primary change (adding rules support). Can you move to a separate PR to make reviewing, merging, reverting and searching commit history easier?